### DISCUSSION TITLE
[TPA #3] Duda calendarización de tarea

### DISCUSSION THREAD

**Discussion by eguiggiaro** [dds-jv/foro#155] [TPA - 2020] [answered]
Hola,

¿Cómo están?

Tenemos una consulta respecto al siguiente punto de la entrega 3 del TPA:

- Deberá crearse una tarea programada que ejecute la validación de egresos (diseñado e implementado en la Entrega 2)  cada cierta cantidad de tiempo.

Usando **Quartz** creamos ciertas entidades que nos permiten planificar la tarea y correrla periódicamente usando una expresión **Cron**. El problema es que esto implicaría que nuestra aplicación esté permanentemente corriendo para ejecutar la tarea periódicamente.

La pregunta es, ¿alcanza con esta implementación? ¿O deberíamos crear la tarea programada desde el OS? En este caso, ¿podemos implementarla en Windows?

Gracias!

**Comment by gastonprieto** [comment, marked as answer]
Buenas!

Por ahora alcanza con eso, seguramente cuando veamos deploy en la última entrega les vamos a pedir que lo cambien a la del sistema operativo. Hablen con su ayudante cuales son las diferencias.

Creo que dado esas condiciones les diria que busquen que creo que Java tiene formas nativas de hacerlo sin meter un framework nuevo, Quartz es superconfiable y tiene un monton de features interesantes de scheduling pero sepan que tal vez es un overhead que después van a descartar.

**Comment by eguiggiaro** [comment]
Genial, gracias!
